### PR TITLE
bring code up to date

### PR DESCRIPTION
**Pull request by Skaruts**
- added self all over
- removed some redundant code

### CODE REVIEW

File: dodgethenim/src/player.nim [hunk 1]
```diff
@@ -1,61 +1,59 @@
...
 
 gdobj Player of Area2D:
   var speed* {.gdExport.} = 400
   var screenSize: Vector2
   var collisionShape2D: CollisionShape2D
+  var animatedSprite: AnimatedSprite
   var body: RigidBody2D
 
-  method init() =
+  method init*() =
```
Comment (Skaruts) [inline dodgethenim/src/player.nim:12]: I exported this method, but I'm not sure it's needed or even what you intended.

(In fact, there seems to be a problem with the init method not being called, at least when you instantiate gdn scripts with `.new()`. The init method also seems to trigger the warning from nim about baseless methods being deprecated.)

**Comment by Skaruts** [comment]
I tinkered with it a bit more and I found that, unlike what I thought, it actually throws an error on every `{.gdExport.}` pragma on fields. I tried adding hints and stuff, but nothing worked. I'm out of ideas. Gonna just wait on pragmagic to see the issue.

Also, on `hud.nim`, it will throw you an error about not finding the file `asyncdispatch`. I couldn't find that anywhere in godot-nim, so it must have been removed (or replaced).

**Comment by zetashift** [comment]
asyncdispatch should refer to: https://nim-lang.org/docs/asyncdispatch.html which is being found on my box. Not sure what's happening there. 

And yeah the gdExport thing isn't great, can't help here sadly so it's gonna be the waiting game.

**Comment by Skaruts** [comment]
Oh, asyncdispatch is a nim thing. So the problem with it is that I imported it as a godotapi file.

**Pull request merged**